src/DynamoCore/Graph/Notes/NoteModel.cs

```diff
@@ -112,11 +120,25 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
             X = helper.ReadDouble("x", 0.0);
             Y = helper.ReadDouble("y", 0.0);
 
+            if(pinnedNode != null)
+                pinnedNode.GUID = helper.ReadGuid("pinnedNode");
```

Comment:
Can we move the code from undoredorecorder here?

Reply:
yes, sure!